route-snapper/src/lib.rs

```diff
@@ -412,9 +437,18 @@ impl JsRouteSnapper {
                     if self.route.waypoints.len() > 1 {
                         self.route.waypoints.remove(idx);
                         self.route.recalculate_full_path(&self.router);
+                        // We're still in a hovering state on the point we just deleted. We may
+                        // want to reset and stop hovering on this point.
+                        self.dont_hover_new_points();
                     }
                 }
             } else {
+                // Clicking an intermediate point shouldn't do anything. It's usually a user error;
```

Comment:
I found a few existingish bugs while testing this out. I would like to brainstorm about how to test the details of the route snapper's interactions... maybe we first need to express/visualize everything happening as a finite state machine, and then brute-force a bunch of possible actions/state transitions from every starting state. There's quite a few states when we take into account these settings (area mode, extend route or not) and edge cases.